### PR TITLE
Story/mov 302 additional meta data

### PR DESCRIPTION
**Pull request by muthenberg**
This PR adds parsing of the new meta data and fixes a few issues I encountered along the way.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -40,50 +40,19 @@ services:
     api:
       container_name: collector_api
       build: .
-      image: cyfacede/collector:2.0.0
-      ports:
-        - "127.0.0.1:8080:8080" # 127.0.0.1 avoids the manipulation of the host's ip-tables
-        - "127.0.0.1:13371:13371"
+      image: cyfacede/collector:2.0.2
+      ports: # It is ok that this entry manipulates the host's ip-tables since this compose file should only be used in dev environments.
```
Comment (hb0) [inline docker-compose.yml]: We should mention this in the Readme then and point to the deployment docker-compose file (if this is publicly available), or else mention that the deployment d-c file is not publicly available.

File: docker-compose.yml [hunk 2]
```diff
@@ -40,50 +40,19 @@ services:
...
+        - "8080:8080"
+        - "13371:13371"
       volumes:
         - ./secrets:/app/secrets
       depends_on:
         - mongo-data
         - mongo-user
         
-    # Prometheus Metrics tool
```
Comment (hb0) [inline docker-compose.yml]: That's awesome that does f**rs are not started by default when I just want to test against the collector, yey!

File: src/main/java/de/cyface/collector/handler/AuthenticationFailureHandler.java [hunk 3]
```diff
@@ -19,23 +19,35 @@
...
+
     @Override
     public void handle(final RoutingContext context) {
+        LOGGER.error("Received failure " + context.failed());
+        LOGGER.error(context.failure());
+
         final boolean closed = context.response().closed();
         final boolean ended = context.response().ended();
         if (!closed && !ended) {
+            LOGGER.error("Failing authentication request with 401 since response was closed: " + closed + ", ended: "
```
Comment (hb0) [inline src/main/java/de/cyface/collector/handler/AuthenticationFailureHandler.java]: I wanted to add such debugging logs too, thanks

File: src/main/java/de/cyface/collector/handler/FormAttributes.java [hunk 4]
```diff
@@ -41,7 +41,27 @@
...
+     */
+    LOCATION_COUNT("locationCount"),
+    /**
+     * The geo location at the beginning of the track represented by the transmitted measurement.
+     */
+    START_LOCATION("startLocation"),
+    /**
+     * The geo location at the end of the track represented by the transmitted measurement.
+     */
+    END_LOCATION("endLocation");
```
Comment (hb0) [inline src/main/java/de/cyface/collector/handler/FormAttributes.java]: What's this for? To be able to load measurements which start/end within a bounding box?

Comment (hb0) [reply src/main/java/de/cyface/collector/handler/FormAttributes.java]: I guess the format of the expected string is defined in the API description you created for the shared API right?

File: src/main/java/de/cyface/collector/verticle/CollectorApiVerticle.java [hunk 5]
```diff
@@ -208,8 +208,8 @@ private void setupRoutes(final String publicKey, final String privateKey, final
                 .failureHandler(new AuthenticationFailureHandler());
         // Set up data collector route
         v2ApiRouter.post("/measurements").handler(BodyHandler.create().setDeleteUploadedFilesOnEnd(false))
-                .handler(JWTAuthHandler.create(jwtAuthProvider)).handler(new MeasurementHandler())
-                .failureHandler(new AuthenticationFailureHandler());
```
Comment (hb0) [inline src/main/java/de/cyface/collector/verticle/CollectorApiVerticle.java]: Was this what you were talking about? That the auth failure handler was not really called so you commented it out?

File: src/test/java/de/cyface/collector/EventBusTest.java [hunk 6]
```diff
@@ -120,23 +128,41 @@ public void handle(Message<String> event) {
...
+                        async.complete();
+                    }
                 });
             }
         });
 
         eventBus.publish(EventBusAddresses.NEW_MEASUREMENT,
-                new Measurement("some_device", "2", "9.0.0", "Pixel 2", Collections.emptyList()));
+                new Measurement("some_device", "2", "9.0.0", "Pixel 2", "4.0.0-alpha1", 200.0, 10,
+                        "lat: 10.0 lon: 10.0, timestamp: 10000", "lat: 12.0 lon: 12.0, timestamp: 12000",
```
Comment (hb0) [inline src/test/java/de/cyface/collector/EventBusTest.java]: now I can sleep well.

File: src/main/java/de/cyface/collector/handler/FormAttributes.java (hunk 4 above)
Comment (muthenberg) [reply src/main/java/de/cyface/collector/handler/FormAttributes.java]: The end location was requested as a feature. I guess it servers for plausibility checking.

The format is specified at least as an example in the API spec. I would prefer not to split this into three properties, but if necessary we will do so in a future version.

File: src/main/java/de/cyface/collector/verticle/CollectorApiVerticle.java (hunk 5 above)
Comment (muthenberg) [reply src/main/java/de/cyface/collector/verticle/CollectorApiVerticle.java]: This causes the system to always return status 401 even for internal server errors or cases where 422 is appropriate. The AuthenticationFailureHandler should only be used for the initial auth call. On other calls the JWTAuthHandler from vertx provides the correct status codes on failure.

**Pull request merged**